controllers/datadogagent/override/global.go

```diff
@@ -6,26 +6,199 @@
 package override
 
 import (
+	"encoding/json"
```

Comment:
I think it'd be nice to have an integration test with an agent that has these "global" options set, but it's something we can leave for a future PR if you prefer.

Reply:
👍 I created a card in the backlog for this.